pkg/routeagent/controllers/route/route.go

```diff
@@ -24,30 +31,51 @@ type Controller struct {
 	clusterID       string
 	objectNamespace string
 
-	submarinerClientSet clientset.Interface
-	clustersSynced      cache.InformerSynced
-	endpointsSynced     cache.InformerSynced
+	submarinerClientSet    clientset.Interface
+	clientSet              *kubernetes.Clientset
+	clustersSynced         cache.InformerSynced
+	endpointsSynced        cache.InformerSynced
+	smRouteAgentPodsSynced cache.InformerSynced
 
 	clusterWorkqueue  workqueue.RateLimitingInterface
 	endpointWorkqueue workqueue.RateLimitingInterface
+	podWorkqueue      workqueue.RateLimitingInterface
 
-	gw      net.IP
-	subnets []string
+	gatewayNodeIP    net.IP
+	localClusterCidr []string
+	localServiceCidr []string
+	remoteSubnets    []string
 
-	link *net.Interface
+	vxlanDevice *vxLanIface
+	vxlanGw     net.IP
+	remoteVTEPs []string
+
+	isGatewayNode bool
+	link          *net.Interface
 }
 
-func NewController(clusterID string, objectNamespace string, link *net.Interface, submarinerClientSet clientset.Interface,
-	clusterInformer informers.ClusterInformer, endpointInformer informers.EndpointInformer) *Controller {
+const VXLAN_IFACE = "vxlan100"
+const VXLAN_PORT = "4800"
```

Comment:
Couldn’t these be integers? That would mean `strconv.Itoa` when they’re needed as strings, but it seems more appropriate to me to have these as integers than strings...